src/common/mmo.hpp

```diff
@@ -721,9 +722,9 @@ struct guild {
 
 struct guild_castle {
 	int castle_id;
-	int mapindex;
-	char castle_name[NAME_LENGTH];
-	char castle_event[EVENT_NAME_LENGTH];
+	uint16 mapindex;
+	std::string castle_name,
+		castle_event;
```

Comment:
I am not sure if this works.
This struct is used in inter-server communication as well. I am not sure if std::string is getting serialized and deserialized properly.
Did you test this?

Reply:
castle_name and castle_event are not in int_guild.cpp nor in the guild_castle table (handled by int_guild.cpp)
```
CREATE TABLE IF NOT EXISTS `guild_castle` (
  `castle_id` int(11) unsigned NOT NULL default '0',
  `guild_id` int(11) unsigned NOT NULL default '0',
  `economy` int(11) unsigned NOT NULL default '0',
  `defense` int(11) unsigned NOT NULL default '0',
  `triggerE` int(11) unsigned NOT NULL default '0',
  `triggerD` int(11) unsigned NOT NULL default '0',
  `nextTime` int(11) unsigned NOT NULL default '0',
  `payTime` int(11) unsigned NOT NULL default '0',
  `createTime` int(11) unsigned NOT NULL default '0',
  `visibleC` int(11) unsigned NOT NULL default '0',
  `visibleG0` int(11) unsigned NOT NULL default '0',
  `visibleG1` int(11) unsigned NOT NULL default '0',
  `visibleG2` int(11) unsigned NOT NULL default '0',
  `visibleG3` int(11) unsigned NOT NULL default '0',
  `visibleG4` int(11) unsigned NOT NULL default '0',
  `visibleG5` int(11) unsigned NOT NULL default '0',
  `visibleG6` int(11) unsigned NOT NULL default '0',
  `visibleG7` int(11) unsigned NOT NULL default '0',
  PRIMARY KEY  (`castle_id`),
  KEY `guild_id` (`guild_id`)
) ENGINE=MyISAM;
```
so there are only used for the mapserver.
I made some tests on local, I have no issue yet

Reply:
Talking about this:
```
int intif_parse_GuildCastleDataLoad(int fd)
{
	return guild_castledataloadack(RFIFOW(fd,2), (struct guild_castle *)RFIFOP(fd,4));
}
```

Reply:
I just reverted the changes in mmo